### PR TITLE
Regex for questionid in God mode too lenient and potentially dangerous #4361

### PR DESCRIPTION
**Pull request by wkurniawan07**
This is NOT a fix. This is to highlight which Strings got erroneously replaced by the lenient regex.
Methodology: change `{*}` to `{***}` for the lenient regex only and see the changes after God mode is run.
Link: #4361

### CODE REVIEW

File: src/test/resources/pages/instructorCourseEnrollError.html [hunk 1]
```diff
@@ -37,7 +37,7 @@
 Team 3 | Frank Hughe
 Team 1 | Black Jack | bjack.gmail.tmt | This student email is invalid
 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa | Robert Downey | rob@email.tmt | This student team name is too long
-Team 2 | {*} | longname@email.tmt | This student name is too long
+Team 2 | {***} | longname@email.tmt | This student name is too long
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorCourseEnrollError.html]: I suppose this is someone's name which doesn't contain space and is 62 characters long or more. Erroneously replaced?

File: src/test/resources/pages/instructorCourseJoinConfirmationHTML.html [hunk 2]
```diff
@@ -130,7 +130,7 @@
                      <br/>
                                        </p>
                   <div class="align-center">
-                     <a class="btn btn-success" href="/page/instructorCourseJoinAuthenticated?key={*}" id="button_confirm">
+                     <a class="btn btn-success" href="/page/instructorCourseJoinAuthenticated?key={***}" id="button_confirm">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorCourseJoinConfirmationHTML.html]: Regkey _without_ `&`.

File: src/test/resources/pages/InstructorCourseDetailsWithSections.html [hunk 3]
```diff
@@ -264,7 +264,7 @@
          <tbody>
             <tr class="student_row" id="student-c0.0">
                <td id="studentphoto-c0.0">
-                  <div class="profile-pic-icon-click align-center" data-link="/page/studentProfilePic?studentemail={*}&courseid={*}&user=CCDetailsUiT.instr2">
+                  <div class="profile-pic-icon-click align-center" data-link="/page/studentProfilePic?studentemail={***}&courseid={***}&user=CCDetailsUiT.instr2">
```
Comment (wkurniawan07) [inline src/test/resources/pages/InstructorCourseDetailsWithSections.html]: The biggest sin of all: student profile picture link. Repeated hundreds of times. This means the regex replace for that in God mode does _nothing_ all these while.

File: src/test/resources/pages/instructorFeedbackResultsDeleteComment.html [hunk 4]
```diff
@@ -350,7 +350,7 @@
                                                             <span class="glyphicon glyphicon-trash glyphicon-primary">
                                                             </span>
                                                          </a>
-                                                         <input name="responseid" type="hidden" value="{*}%CFResultsUiT.instr@gmail.tmt%%GENERAL%"/>
+                                                         <input name="responseid" type="hidden" value="{***}%CFResultsUiT.instr@gmail.tmt%%GENERAL%"/>
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackResultsDeleteComment.html]: Response IDs. There are regex replace for this, but works _only if_ it's in the form of `questionid%email%email`. `%GENERAL%` doesn't fit the email regex, neither does `Team Name`.

**Comment by wkurniawan07** [comment]
And... it's even more surprising to see that out of all the things that this regex replaces, _none of them are questionid_.

**Comment by kanghj** [comment]
@thyageshm

**Comment by wkurniawan07** [comment]
Don't worry, I'm fixing this as well.

**Pull request closed without merging**